sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -186,66 +173,30 @@ internal static string GetOperationName(Activity activity, ref AzMonList MappedT
             return activity.DisplayName;
         }
 
-        internal static string GetNewSchemaOperationName(Activity activity, string? url, ref AzMonList MappedTags)
+        internal static string GetOperationNameV2(Activity activity, ref AzMonList MappedTags)
         {
             var httpMethod = AzMonList.GetTagValue(ref MappedTags, SemanticConventions.AttributeHttpRequestMethod)?.ToString();
```

Comment:
`http.request.method` is a required attribute. We should avoid the condition check for it and use GetValues for `SemanticConventions.AttributeHttpRequestMethod` and `SemanticConventions.AttributeHttpRoute` together. It will reduce one loop for every span.

Reply:
I did look into this earlier. `GetTagValue` is faster than `GetTagValues`. Plus `GetTagValues` have array allocation. Need to investigate more on why is `GetTagValues` slower.

Reply:
The `GetTagValue` function returns an object. If you call it twice, it allocates two objects at different locations. However, the `GetTagValues` function accomplishes the same result by storing the two objects in a contiguous location, which is considered better.

I remember getting benchmarks before merging `GetTagValues`. If the list values grow, `GetTagValues` does a better job.

Reply:
Here is what I am getting
![image](https://github.com/Azure/azure-sdk-for-net/assets/17655898/e8a8eae6-fb2b-4b36-a82e-5bd471b57104)

We only have benchmark for [GetTagValue](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Benchmarks/TagObjectsGetValuesBenchmarks.cs) which looks closer to what I am getting. Will sync offline on this

Reply:
This is not a blocker for this PR, we will sync offline.